### PR TITLE
chore: improve project tooling

### PR DESCRIPTION
**Pull request by pedronauck**
## Tasks

This pull request makes some changes to improve precise details on our tooling

### `.lintstagedrc`
- [x] Change to `prettier --write` and `eslint --fix`. So, instead of just checking on husky, it will format and fix the files before committing.
- [x] Add other formats on `lintstaged` to run `prettier`

### `.prettierrc`
- [x] Setup prettier to format all types of files, since we have a bunch new of them and this configuration  came from `fuels-ts` that's a totally different type of project

### `.vscode` folder
- [x] Add a recommendation of a new extension that is a [Jest runner](https://marketplace.visualstudio.com/items?itemName=firsttris.vscode-jest-runner) inside VSCode. I used this extension on `fuel-ui` and it's capable to debug using VSCode and Jest, it's very useful for debugging without console.log on tests, and this extension is way better than the with most downloads on VSCode extensions store.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,7 +1,11 @@
 {
   "editor.defaultFormatter": "esbenp.prettier-vscode",
   "prettier.prettierPath": "./node_modules/prettier",
+  "editor.formatOnPaste": false,
+  "editor.formatOnType": false,
   "editor.formatOnSave": true,
+  "editor.formatOnSaveMode": "file",
+  "files.autoSave": "onFocusChange",
```
Comment (LuizAsFight) [inline .vscode/settings.json:8]: I think we don't need this line... saving should be intentionally fired manually

what you think @luizstacio  ??

I use cmd+shift+s when I want to save all files

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
minor comment

File: .vscode/settings.json (hunk 1 above)
Comment (pedronauck) [reply .vscode/settings.json:8]: Hm, but in which way this can change the `ctrl+z/ctrl+shift+z`? I didn't get it 🤔

Comment (LuizAsFight) [reply .vscode/settings.json:8]: it helps identifying what files you changed last, then use undo/redo if needed...
if applying this change everything will be always saved when change file right ?

Comment (LuizAsFight) [reply .vscode/settings.json:8]: also many times we need to change a lot of files to get a change done... if autoSave is enabled, live reload will break app you're running a lot of times...

saving should be manual, if want can execute `cmd+shift+s` to save all files, that's what I do when complete changes in multiple files

Comment (pedronauck) [reply .vscode/settings.json:8]: I always used this personal setting on my VSCode and I never have problems with this. Some extensions recommend this type of autosave:
CleanShot 2023-01-26 at 22 58 04

This is good to ensure that, if something VSCode you don't lose what you did I guess also.
About the live reloading, I don't think this is a point.

But idk, is a point of view, I can remove it, no problem.

Comment (pedronauck) [reply .vscode/settings.json:8]: Let's see whats the rest think about @tomiiide @luizstacio

Comment (LuizAsFight) [reply .vscode/settings.json:8]: we can always add later if @FuelLabs/frontend think it's good.. I personally don't like it

can wait for other to answer 

or just remove for now and add later in other PR, then we can quickly approve this one

Comment (LuizAsFight) [reply .vscode/settings.json:8]: thing with `ctrl+z / ctrl+shift+z` is that if the file is unsaved, you know how many `undo` you need to get to the previous saved(valid) one

Comment (LuizAsFight) [reply .vscode/settings.json:8]: ask for re-review when this is solved

Comment (LuizAsFight) [reply .vscode/settings.json:8]: sounds good @pedronauck

Comment (luizstacio) [reply .vscode/settings.json:8]: I also don't like auto-save, we probably should remove it from the project itself.

File: .changeset/silent-sloths-flow.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
-"@fuel-wallet/types": minor
-"@fuel-wallet/sdk": minor
+'@fuel-wallet/types': minor
+'@fuel-wallet/sdk': minor
```
Comment (luizstacio) [inline .changeset/silent-sloths-flow.md:3]: This will not be an issue for the changesets?

Comment (pedronauck) [reply .changeset/silent-sloths-flow.md:3]: why?

File: .vscode/settings.json (hunk 1 above)
Comment (pedronauck) [reply .vscode/settings.json]: removed

File: .changeset/silent-sloths-flow.md (hunk 2 above)
Comment (luizstacio) [reply .changeset/silent-sloths-flow.md:3]: Is more a question as I don't know how changeset cli parsers

**Review by luizstacio — DISMISSED** [review]

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,7 +1,6 @@
 {
   "editor.defaultFormatter": "esbenp.prettier-vscode",
   "prettier.prettierPath": "./node_modules/prettier",
-  "editor.formatOnSave": true,
```
Comment (LuizAsFight) [inline .vscode/settings.json:4]: it's good to keep `formatOnSave` ,

my prev comment was about `autoSave` instead

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

**Review by LuizAsFight — APPROVED** [review]
great work!

**Pull request merged**